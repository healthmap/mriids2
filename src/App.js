import React, { Component } from "react";
import { connect } from "react-redux";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { theme } from "./assets/theme";
import { fetchEbolaData, fetchEbolaDataCombined } from "./actions/ebola";
import { fetchCovidCaseCounts, fetchCovidDeathCounts } from "./actions/covid";
import SnapshotMap from "./components/SnapshotMap";
import EbolaRiskMap from "./containers/EbolaRiskMap";
import Team from "./components/Team";
import About from "./components/About";
import Header from "./components/Header";
import Sidebar from "./components/Sidebar";
import ChartComponent from "./components/ChartComponent";
import DateRange from "./components/DateRange";
import { StyledAppContainer } from "./styles";

class App extends Component {
  componentDidMount() {
    this.props.fetchEbolaData();
    this.props.fetchEbolaDataCombined();
    this.props.fetchCovidCaseCounts();
    this.props.fetchCovidDeathCounts();
  }

  renderHomePageComponents = () => {
    // When we are displaying the EbolaRiskMap, we want this to display on the entire page.
    if (
      this.props.filters.view === "risk" &&
      this.props.filters.outbreak === "Ebola Outbreak"
    ) {
      return <EbolaRiskMap />;
    } else {
      // If we are not displaying the EbolaRiskMap, we want to display the SnapshotMap, ChartComponent, and DateRange.
      return (
        <>
          <SnapshotMap />
          <ChartComponent />
          <DateRange />
        </>
      );
    }
  };

  render() {
    return (
      <ThemeProvider theme={theme}>
        <Router>
          <StyledAppContainer>
            <Header />
            <Switch>
              <Route exact path="/">
                <Sidebar />
                {this.renderHomePageComponents()}
              </Route>
              <Route exact path="/about">
                <About />
              </Route>
              <Route exact path="/team">
                <Team />
              </Route>
            </Switch>
          </StyledAppContainer>
        </Router>
      </ThemeProvider>
    );
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters,
});

const mapDispatchToProps = (dispatch) => ({
  fetchEbolaData: () => dispatch(fetchEbolaData()),
  fetchEbolaDataCombined: () => dispatch(fetchEbolaDataCombined()),
  fetchCovidCaseCounts: () => dispatch(fetchCovidCaseCounts()),
  fetchCovidDeathCounts: () => dispatch(fetchCovidDeathCounts()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
