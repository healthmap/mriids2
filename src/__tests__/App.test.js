import React from "react";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import App from "../App";

const mockStore = configureStore([thunk]);

describe("Tests for the App component", () => {
  let store;
  let component;

  const initialState = {
    ebolaData: {
      isFetching: 0,
      data: {},
      error: {},
    },
    ebolaDataCombined: {
      isFetching: 0,
      data: {},
      error: {},
    },
    riskData: {
      isFetching: 0,
      data: {},
      error: {},
    },
  };

  beforeEach(() => {
    store = mockStore(initialState);

    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
  test("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
