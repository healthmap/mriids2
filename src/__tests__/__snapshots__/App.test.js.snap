// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the App component should render with given state from Redux store 1`] = `
.c8 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  padding-left: 21.6rem;
  height: 50rem;
  overflow: hidden;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  font-size: 1.4rem;
  line-height: 2;
}

.c3 li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 li a,
.c3 li a:visited,
.c3 li a:hover,
.c3 li a:focus,
.c3 li a:active {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 2rem;
  color: #000;
  font-weight: bold;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 li a.is-active,
.c3 li a:visited.is-active,
.c3 li a:hover.is-active,
.c3 li a:focus.is-active,
.c3 li a:active.is-active {
  background-color: #d4d4d4;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 100%;
  height: 5.6rem;
  box-shadow: 0 10px 12px -8px rgba(0,0,0,0.1);
  z-index: 3;
}

.c1 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 21.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 img {
  width: auto;
  height: 2.4rem;
}

.c6 {
  display: block;
  width: 100%;
  padding: 1.5rem;
  font-size: 1.4rem;
  font-weight: bold;
  line-height: 1.3;
  border: 0;
  background-image: url("data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4NCjwhLS0gR2VuZXJhdG9yOiBBZG9iZSBJbGx1c3RyYXRvciAxNi4wLjQsIFNWRyBFeHBvcnQgUGx1Zy1JbiAuIFNWRyBWZXJzaW9uOiA2LjAwIEJ1aWxkIDApICAtLT4NCjwhRE9DVFlQRSBzdmcgUFVCTElDICItLy9XM0MvL0RURCBTVkcgMS4xLy9FTiIgImh0dHA6Ly93d3cudzMub3JnL0dyYXBoaWNzL1NWRy8xLjEvRFREL3N2ZzExLmR0ZCI+DQo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IkxheWVyXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4Ig0KCSB3aWR0aD0iNjEycHgiIGhlaWdodD0iNzkycHgiIHZpZXdCb3g9IjAgMCA2MTIgNzkyIiBlbmFibGUtYmFja2dyb3VuZD0ibmV3IDAgMCA2MTIgNzkyIiB4bWw6c3BhY2U9InByZXNlcnZlIj4NCjx0aXRsZT5hcnJvd3M8L3RpdGxlPg0KPHBhdGggZmlsbD0iI0ZGRkZGRiIgZD0iTTIyMS42NTIsMzY5Ljg2NGw4NC43NDUtMTE4LjAwOGw4My45NTEsMTE4LjAwOEgyMjEuNjUyeiBNMzkwLjM0OCw0MjIuMTM2bC04My45NTEsMTE4LjAwOGwtODQuNzQ1LTExOC4wMDgNCglIMzkwLjM0OHoiLz4NCjwvc3ZnPg0K");
  background-position: right 0px center;
  background-repeat: no-repeat;
  background-size: auto 80%;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c6:focus {
  outline: none;
  border-color: transparent;
  box-shadow: 1px 1px 6px 0px rgba(39,124,218,0.2);
}

.c4 {
  position: absolute;
  top: 5.6rem;
  left: 0;
  bottom: 0;
  width: 21.6rem;
  background-color: #efefef;
  overflow: auto;
  z-index: 2;
}

.c4 h4 {
  padding-top: 1em;
}

.c7 select {
  color: #fff;
  background-color: #e43b46;
}

.c7 option {
  color: #fff;
  background-color: #e43b46;
}

.c5 select {
  color: #fff;
  background-color: #000;
}

.c5 option {
  color: #fff;
  background-color: #000;
}

.c9 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  padding-left: 21.6rem;
  height: 28rem;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  width: 100vw;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="mriids"
          src="/mriids.svg"
        />
      </a>
    </span>
    <ul
      className="c3"
    >
      <li>
        <a
          aria-current="page"
          className="is-active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Outbreak
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/team"
          onClick={[Function]}
        >
          Team
        </a>
      </li>
    </ul>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <select
        className="c6"
        name="location"
        onChange={[Function]}
        value="All"
      >
        <option
          value="All"
        >
          All
        </option>
        <option
          value="Guinea"
        >
          Guinea
        </option>
        <option
          value="Liberia"
        >
          Liberia
        </option>
        <option
          value="Sierra Leone"
        >
          Sierra Leone
        </option>
      </select>
    </div>
    <div
      className="c7"
    >
      <select
        className="c6"
        name="outbreak"
        onChange={[Function]}
        value="Ebola Outbreak"
      >
        <option
          value="Ebola Outbreak"
        >
          Ebola Outbreak
        </option>
        <option
          value="COVID 19"
        >
          COVID 19
        </option>
      </select>
    </div>
  </div>
  <div
    className="c8"
  >
    <div
      style={
        Object {
          "cursor": "grab",
          "height": 0,
          "position": "relative",
          "width": 0,
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          className=""
          style={
            Object {
              "height": "100%",
              "overflow": "hidden",
              "position": "absolute",
              "visibility": "inherit",
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "height": 0,
              "overflow": "visible",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="c9"
  >
    <div>
      Loading Chart
    </div>
  </div>
</div>
`;
